src/components/side_panel/chart/building_blocks/general_series/general_series_editor.ts

```diff
@@ -0,0 +1,123 @@
+import { Component, useState } from "@odoo/owl";
+import { getNthColor, toHex } from "../../../../../helpers";
+import { getDefinedAxis } from "../../../../../helpers/figures/charts";
+import { _t } from "../../../../../translation";
+import {
+  ChartWithAxisDefinition,
+  DispatchResult,
+  SpreadsheetChildEnv,
+  UID,
+} from "../../../../../types";
+import { SidePanelCollapsible } from "../../../components/collapsible/side_panel_collapsible";
+import { RoundColorPicker } from "../../../components/round_color_picker/round_color_picker";
+import { Section } from "../../../components/section/section";
+
+interface Props {
+  figureId: UID;
+  definition: ChartWithAxisDefinition;
+  updateChart: (figureId: UID, definition: Partial<ChartWithAxisDefinition>) => DispatchResult;
+  canChangeVerticalAxis?: boolean;
+}
+
+export class GeneralSeriesEditor extends Component<Props, SpreadsheetChildEnv> {
+  static template = "o-spreadsheet-GeneralSeriesEditor";
+  static components = {
+    SidePanelCollapsible,
+    Section,
+    RoundColorPicker,
+  };
+  static props = {
+    figureId: String,
+    definition: Object,
+    updateChart: Function,
+    canChangeVerticalAxis: { type: Boolean, optional: true },
+    slots: { type: Object, optional: true },
+  };
+
+  private state = useState({ index: -1 });
+
+  getDataSeries() {
+    const runtime = this.env.model.getters.getChartRuntime(this.props.figureId);
```

Comment:
Do you think it's possible to not use the runtime in this component ? It kinda make the side panel depend on ChartJS, which isn't a problem per se, but not really amazing